### PR TITLE
User and Session Database Adapter improvements 

### PR DESCRIPTION
**Pull request by roosnic1**
WPC-449. This PR adds fields and methods to the user and session models. With those it will be possible for users of we.publish to implement their own authentication solution in the website.

### CODE REVIEW

File: packages/api/src/db/user.ts [hunk 1]
```diff
@@ -45,13 +50,6 @@ export interface UserSubscriptionFilter {
   readonly autoRenew?: boolean
 }
 
-export interface UserSubscriptionFilter {
```
Comment (roosnic1) [inline packages/api/src/db/user.ts:48]: removed the interface because it was duplicated.

File: packages/api/src/db/user.ts [hunk 2]
```diff
@@ -76,14 +74,6 @@ export interface UserSubscriptionPeriodInput {
   readonly invoiceID: string
 }
 
-export interface UserSubscriptionPeriodInput {
```
Comment (roosnic1) [inline packages/api/src/db/user.ts:79]: removed the interface because it was duplicated.

File: packages/api-db-mongodb/src/db/schema.ts [hunk 3]
```diff
@@ -100,11 +101,14 @@ export interface DBUser {
   modifiedAt: Date
 
   email: string
+  emailVerified: Date | null
```
Comment (penk85) [inline packages/api-db-mongodb/src/db/schema.ts:104]: Would it be better to have an indication in the naming that this is a date? `emailVerifiedAt` or something?

File: packages/api-db-mongodb/src/migration.ts [hunk 4]
```diff
@@ -593,6 +593,26 @@ export const Migrations: Migration[] = [
         }
       )
     }
+  },
+  {
+    //  Rename street field in address to address
```
Comment (penk85) [inline packages/api-db-mongodb/src/migration.ts:598]: I think you meant to change this

**Review by penk85 — CHANGES_REQUESTED** [review]
Left some comments

File: packages/api-db-mongodb/src/db/session.ts [hunk 5]
```diff
@@ -99,6 +99,32 @@ export class MongoDBSessionAdapter implements DBSessionAdapter {
     return null
   }
 
+  async extendUserSessionByToken(token: string): Promise<OptionalUserSession> {
```
Comment (bishoymelek) [inline packages/api-db-mongodb/src/db/session.ts:102]: I think it's so helpful to keep users logged-in depending on specific condition(s) 👍

File: packages/api-db-mongodb/src/db/user.ts [hunk 6]
```diff
@@ -475,6 +520,57 @@ export class MongoDBUserAdapter implements DBUserAdapter {
...
+
+    const {value} = await this.users.findOneAndUpdate(
+      {_id: userID},
+      {
+        $set: {
+          modifiedAt: new Date(),
+          oauth2Accounts: accounts
+        }
+      },
+      {returnOriginal: false}
```
Comment (bishoymelek) [inline packages/api-db-mongodb/src/db/user.ts:538]: I think after a while we have to update this one as it's deprecated [see here](https://stackoverflow.com/a/67909382/7254406)

File: packages/api-db-mongodb/src/migration.ts [hunk 7]
```diff
@@ -593,6 +593,26 @@ export const Migrations: Migration[] = [
...
+          emailVerified: {$exists: false}
+        },
+        {$set: {emailVerified: null}}
+      )
+
+      await users.updateMany(
+        {
+          oauth2Accounts: {$exists: false}
+        },
+        {$set: {oauth2Accounts: []}}
```
Comment (bishoymelek) [inline packages/api-db-mongodb/src/migration.ts:613]: can't you add this attr to the upper mutation? why two?

**Review by bishoymelek — CHANGES_REQUESTED** [review]
@roosnic1 really helpful feature 🎉 , Just one comment and the `emailVerified` field @penk85 commented

File: packages/api-db-mongodb/src/db/user.ts (hunk 6 above)
Comment (roosnic1) [reply packages/api-db-mongodb/src/db/user.ts:538]: True but that will happen in the general refactoring of core.

File: packages/api-db-mongodb/src/migration.ts (hunk 7 above)
Comment (roosnic1) [reply packages/api-db-mongodb/src/migration.ts:613]: because it's should only set the field it it doesn't exist.

**Comment by roosnic1** [comment]
Closing PR because the branch name is wrong.

**Pull request closed without merging**